spec/lib/avalon/build_utils_spec.rb

```diff
@@ -0,0 +1,166 @@
+    it 'Detects 2 part version correctly (not supported)' do
+      version = "1.2"
+      contents_arr = get_contents_array(version)
+      expect(Utils.detect_version(contents_arr)).to eq('')
+    end
+
+    it 'Detects 1 part version correctly (not supported)' do
+      version = "1"
+      contents_arr = get_contents_array(version)
+      expect(Utils.detect_version(contents_arr)).to eq('')
+    end
+
+    it 'Ignores commented lines in config file contents' do
+      version = "3.14.1"
+      contents_arr = []
+      contents_arr.push "# VERSION = '2.0.1'"
+      contents_arr.concat get_contents_array(version)
+      expect(Utils.detect_version(contents_arr)).to eq('3.14.1')
+    end
+
+    describe '#get_tags' do
+      it 'Returns an empty array if no branch nor top_level options' do
+        version = "1.2.3"
+        expect(Utils.get_tags(version)).to eq([])
+      end
+      it 'Outputs correct tags for a 3-part version and top_level true' do
+        version = "1.2.3"
+        options = {"top_level": true}
+        expect(Utils.get_tags(version, options)).to eq(["1.2.3"])
+      end
+
+      it 'SPLITS and outputs correct tags for a 3-part version and no additional tags' do
+        version = "1.2.3"
+        options = {"top_level": true, "split": true}
+        expect(Utils.get_tags(version, options)).to eq(["1", "1.2", "1.2.3"])
+      end
+
+      it 'Outputs correct tags for a 3-part version and a branch tag' do
+        version = "3.2.1"
+        options = {"split": false, "branch": "staging"}
+        expect(Utils.get_tags(version, options)).to eq(["3.2.1-staging", "staging"])
+      end
+
+      it 'SPLITS and Outputs correct tags for a 3-part version and a branch tag TOP LEVEL production branch' do
+        version = "3.2.1"
+        options = {"split": true, "branch": "production", "top_level": true}
+        expect(Utils.get_tags(version, options)).to eq(["3-production", "3.2-production", "3.2.1-production", "production"])
+      end
+
+      it 'SPLITS and Outputs correct tags for a 3-part version and a branch tag NOT TOP LEVEL production branch' do
+        version = "3.2.1"
+        options = {"split": true, "branch": "production", "top_level": false}
+        expect(Utils.get_tags(version, options)).to eq(["3-production", "3.2-production", "3.2.1-production", "production"])
+      end
+
+      it 'Outputs correct tags for a 3-part version and a branch tag TOP LEVEL production branch' do
+        version = "3.2.1"
+        branch = "production"
+        options = {"split": false, "branch": "production", "top_level": true}
+        expect(Utils.get_tags(version, options)).to eq(["3.2.1-production", "3.2.1", "production"])
+      end
+
+      it 'SPLITS and outputs correct tags for a 3-part version and a branch tag' do
+        version = "3.2.1"
+        options = {"split": true, "branch": "staging"}
+        expect(Utils.get_tags(version, options)).to eq(["3-staging", "3.2-staging", "3.2.1-staging", "staging"])
+      end
```

Comment:
Is this a duplicate of the test on lines 117-121 given that `top_level` defaults to `false`?

Reply:
Yes, I also changed the branch name in the prior instance to use `staging` since toplefel really should not be used with anything other than `production`